rosidl_runtime_py/convert.py

```diff
@@ -143,9 +143,9 @@ def to_string(val, field_type=None):
         return r
     result = ''
 
-    # We rely on __slots__ retaining the order of the fields in the .msg file.
-    for field_name, field_type in zip(msg.__slots__, msg.SLOT_TYPES):
-        value = getattr(msg, field_name)
+    # We rely on the get_fields_and_field_types() method to retrieve the fields in the .msg file.
+    for field_name, field_type in msg.get_fields_and_field_types().items():
+        value = getattr(msg, '_' + field_name)
```

Comment:
Actually, I'm confused about this.

Looking at the generated code, and testing out locally, it should be totally correct to use `getattr(msg, field_name)`.  We generate `property` methods for all of them.  That is, the following works when I try locally:

```
>>> from std_srvs.srv import SetBool_Request
>>> x = SetBool_Request()
>>> x.data
False
>>> getattr(x, 'data')
False
```

So why is this needed?

(same question below)

Reply:
You are absolutely right, there is no need for adding the "_" before the attribute name since we have the `property` methods. When I was debugging why other packages were failing CI I notices that and made the changes to have the exact same values that we were retrieving by using `__slots__`. However, the real fix is the one on the line 185.